packages/promise-tracker/lib/index.ts

```diff
@@ -0,0 +1,104 @@
+const _outstandingPromiseInstanceMap = new WeakMap<Object, Set<Promise<any>>>();
+const _outstandingPromiseMap = new Map<Promise<any>, boolean>();
+
+export function tracked<T extends Object>(
+  target: T,
+  key: string | Symbol,
+  descriptor: PropertyDescriptor
+) {
+  if (typeof descriptor.value === "function") {
```

Comment:
should @tracked be allows on non-functions and non-promises at all?

Reply:
No need for it so far. Also at the moment it's strictly a method decorator. I thought I addressed that in the README, but yeah - if people need it to decorate something other than a method, they'll need to implement that.

Reply:
I'll also see if I can't make it throw when decorating something other than a method.

Reply:
> I'll also see if I can't make it throw when decorating something other than a method.

Changes that I'll be pushing shortly make it so that I get compiler errors when trying to decorate anything other than a method. I've still added a `throw` check just for good measure, however.